src/plugins/vis_type_timeseries/public/application/components/aggs/field_select.js

```diff
@@ -64,7 +70,7 @@ function FieldSelectUi({
           };
         }
 
-        if (value === item.value) {
+        if (extractTimefieldLabel(value) === item.label) {
```

Comment:
This causes a regression for saved objects:
a saved object stores a field as a value string, but the new one has `timeFieldObject` shape.  
you could check the regression in `[eCommerce] Promotion Tracking` sample data vis: fields are not selected anymore